apps/ensadmin/src/components/recent-registrations/components.tsx

```
@@ -135,18 +136,19 @@ const NAME_WRAPPER_ADDRESS = "0xd4416b13d2b3a9abae7acd5d6c2bbdbe25686401";
  * Determines the true owner of a domain.
  * If the owner is the NameWrapper contract, returns the wrapped owner instead.
  *
- * @param owner The owner address
- * @param wrappedOwner The wrapped owner address (optional)
- * @returns The true owner address
+ * @param registration - Extended data associated with Registration event
  */
-function getTrueOwner(owner: { id: Address }, wrappedOwner?: { id: Address }) {
+function getTrueOwner(registration: Registration) {
   // Only use wrapped owner if the owner is the NameWrapper contract
-  if (wrappedOwner?.id && isAddressEqual(owner.id, NAME_WRAPPER_ADDRESS)) {
-    return getAddress(wrappedOwner.id);
+  if (
+    registration.registration.wrappedOwner &&
+    isAddressEqual(registration.registration.owner, NAME_WRAPPER_ADDRESS)
```

FYI: Here what we're really doing is we're checking if the `ownerInRegistry` is equal to the `NAME_WRAPPER_ADDRESS`. If so, then `ownerInNameWrapper` MUST be defined (otherwise we should throw an exception as our expectations are violated) and the `ownerInNameWrapper` should be considered the effective owner.